pkg/plugin/restore_pvc_action_plugin.go

```diff
@@ -140,6 +142,31 @@ func (p *NewPVCRestoreItemAction) Execute(input *velero.RestoreItemActionExecute
 		return nil, errors.WithStack(err)
 	}
 
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	getPvc, err := kubeClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
+	if err != nil && !apierrs.IsNotFound(err) && !apierrs.IsAlreadyExists(err) {
+		p.Log.Errorf("Error occurred when trying to get PVC %s/%s: %v", pvc.Namespace, pvc.Name, err)
+		return nil, errors.WithStack(err)
+	} else if getPvc != nil && getPvc.Namespace == pvc.Namespace && getPvc.Name == pvc.Name {
+		// NOTE: If PVC does not exist, "Get" may return an empty PersistentVolumeClaim object, not nil; so we need to check Namespace/Name match here
+		// NOTE: Need to skip it here in the plugin to avoid hanging as Velero skips restoring already existing resources
+		p.Log.Warnf("Skipping PVCRestoreItemAction for PVC %s/%s since it already exists.", pvc.Namespace, pvc.Name)
+		return &velero.RestoreItemActionExecuteOutput{
+			UpdatedItem: input.Item,
+		}, nil
+	} else if err != nil && apierrs.IsAlreadyExists(err) {
```

Comment:
What is the case where a GET API returns `IsAlreadyExists` error? I wonder if it is possible.

Reply:
Removed this.  I think this happens when creating a resource.